packages/netlify-cms-core/src/components/Collection/Entries/EntriesCollection.js

```diff
@@ -31,10 +31,10 @@ class EntriesCollection extends React.Component {
     }
   }
 
-  componentWillReceiveProps(nextProps) {
+  componentDidUpdate(prevProps) {
     const { collection, loadEntries } = this.props;
-    if (nextProps.collection !== collection) {
-      loadEntries(nextProps.collection);
+    if (collection !== prevProps.collection) {
+      loadEntries(collection);
```

Comment:
[Side effects on props change should be in `componentDidUpdate`](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#side-effects-on-props-change)